tests_pytest/_fw_tests_common_base/base_graph_with_fusion_metadata_test.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2025 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Callable, Any
+
+import copy
+
+import abc
+
+import pytest
+
+
+import model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema as schema
+from model_compression_toolkit.core import QuantizationConfig
+from model_compression_toolkit.core.common import BaseNode
+from model_compression_toolkit.core.common.graph.edge import EDGE_SOURCE_INDEX, EDGE_SINK_INDEX
+from model_compression_toolkit.core.graph_prep_runner import graph_preparation_runner
+
+
+class BaseGraphWithFusingMetadataTest(abc.ABC):
```

Comment:
additional integration test to consider: successful graph update with:
1. with no effect on any fusion in the graph (verify that the fusion validation does not fail us unintentionally)
2. with effect on the graph that is addressed in the substitution (if there such)

Reply:
Regard 1 - I agree.
Regard 2 - These are the tests below. Or did I misunderstand you?